paasta_tools/cli/cmds/status.py

```diff
@@ -299,33 +305,357 @@ def print_marathon_status(
+        autoscaling_info.current_utilization = "{:.1f}%".format(
+            autoscaling_info.current_utilization * 100
+        )
     else:
-        deploy_status = "NotRunning"
+        autoscaling_info.current_utilization = "Exception"
 
+    if autoscaling_info.target_instances is None:
+        autoscaling_info.target_instances = "Exception"
+
+    headers = list(autoscaling_fields_to_headers.values())
+    row = [
+        str(getattr(autoscaling_info, field)) for field in autoscaling_fields_to_headers
+    ]
+    table = [f"  {line}" for line in format_table([headers, row])]
+    output.extend(table)
+    return output
+
+
+def marathon_mesos_status_human(
+    error_message,
+    running_task_count,
+    expected_instance_count,
+    running_tasks,
+    non_running_tasks,
+):
+    if error_message:
+        return [PaastaColors.red(error_message)]
+
+    output = []
     output.append(
-        "    {}".format(
-            status_marathon_job_human(
-                service=service,
-                instance=instance,
-                deploy_status=deploy_status,
-                desired_app_id=marathon_status.app_id,
-                app_count=marathon_status.app_count,
-                running_instances=marathon_status.running_instance_count,
-                normal_instance_count=marathon_status.expected_instance_count,
+        marathon_mesos_status_summary(running_task_count, expected_instance_count)
+    )
+
+    if running_tasks or non_running_tasks:
+        output.append("  Running Tasks:")
+        running_tasks_table = create_mesos_running_tasks_table(running_tasks)
+        output.extend([f"    {line}" for line in running_tasks_table])
+
+        output.append(PaastaColors.grey("  Non-running Tasks:"))
+        non_running_tasks_table = create_mesos_non_running_tasks_table(
+            non_running_tasks
+        )
+        output.extend([f"    {line}" for line in non_running_tasks_table])
+
+    return output
+
+
+def create_mesos_running_tasks_table(running_tasks):
+    rows = []
+    table_header = [
+        "Mesos Task ID",
+        "Host deployed to",
+        "Ram",
+        "CPU",
+        "Deployed at what localtime",
+    ]
+    rows.append(table_header)
+    for task in running_tasks or []:
+        mem_string = get_mesos_task_memory_string(task)
+        cpu_string = get_mesos_task_cpu_string(task)
+        deployed_at = datetime.fromtimestamp(task.deployed_timestamp)
+        deployed_at_string = "{} ({})".format(
+            deployed_at.strftime("%Y-%m-%dT%H:%M"), humanize.naturaltime(deployed_at)
+        )
+
+        rows.append(
+            [task.id, task.hostname, mem_string, cpu_string, deployed_at_string]
+        )
+        rows.extend(format_tail_lines_for_mesos_task(task.tail_lines, task.id))
+
+    return format_table(rows)
+
+
+def get_mesos_task_memory_string(task):
+    if task.rss.value is None or task.mem_limit.value is None:
+        return task.mem_limit.error_message
```

Comment:
Is this guaranteed to be set in this case if either are these are none?

Reply:
Good point -- I'll make it look at both `error_message` fields.  They should both be `None` if one of them is because they they both come from the task stats on the underlying Mesos task, but that's not necessarily something on this side of the API should know about.